src/VisualStudio/Core/Def/ValueTracking/ValueTrackingTree.xaml.cs

```diff
@@ -32,9 +32,97 @@ public ValueTrackingTree(ValueTrackingTreeViewModel viewModel)
             InitializeComponent();
         }
 
-        private void ValueTrackingTreeView_SelectedItemChanged(object sender, RoutedPropertyChangedEventArgs<object> e)
+        private void ValueTrackingTreeView_PreviewKeyDown(object sender, KeyEventArgs e)
         {
-            _viewModel.SelectedItem = (TreeItemViewModel)e.NewValue;
+            e.Handled = e.Handled || e.Key switch
+            {
+                Key.Down => TrySelectItem(GetNextItem(expandNode: false), navigate: false),
+                Key.Up => TrySelectItem(GetPreviousItem(), navigate: false),
+                Key.F8 => e.KeyboardDevice.Modifiers == ModifierKeys.Shift ? TrySelectItem(GetPreviousItem(), navigate: true) : TrySelectItem(GetNextItem(expandNode: true), navigate: true),
+                Key.Enter => TrySelectItem(ValueTrackingTreeView.SelectedItem as TreeViewItemBase, navigate: true),
+                Key.Right => TrySetExpanded(ValueTrackingTreeView.SelectedItem as TreeViewItemBase, true),
+                Key.Left => TrySetExpanded(ValueTrackingTreeView.SelectedItem as TreeViewItemBase, false),
+                Key.Space => TryToggleExpanded(ValueTrackingTreeView.SelectedItem as TreeViewItemBase),
+                _ => false
+            };
+
+            // Local Functions
+
+            static bool TrySelectItem(TreeViewItemBase? node, bool navigate)
+            {
+                if (node is null)
+                {
+                    return false;
+                }
+
+                SelectItem(node, navigate);
+                return true;
+            }
+
+            static bool TrySetExpanded(TreeViewItemBase? node, bool expanded)
+            {
+                if (node is null)
+                {
+                    return false;
+                }
+
+                node.IsNodeExpanded = expanded;
+                return true;
+            }
+
+            static bool TryToggleExpanded(TreeViewItemBase? node)
+                => TrySetExpanded(node, node is null ? false : !node.IsNodeExpanded);
+        }
+
+        private void ValueTrackingTreeView_MouseDoubleClick(object sender, MouseButtonEventArgs e)
+        {
+            if (sender is TreeViewItemBase viewModel)
+            {
+                SelectItem(viewModel, true);
```

Comment:
Shouldn't this also expand, based on your table?

Reply:
Actually, I think the table is wrong here. Imo clicking on an item shouldn't expand at all. Open to feedback on that behavior though

Reply:
Updated table to reflect this behavior as written

Reply:
As the person who made double click on a project file open the editor, and not expand/collapse, I get the logic of it and support you, but you should probably expect feedback too :)

Reply:
I believe double-clicking in Test Explorer would do an expansion. And I just know solution explorer does a different thing😮